addons/misra.py

```diff
@@ -700,37 +716,30 @@ def misra_3_1(self, rawTokens):
 
     def misra_4_1(self, rawTokens):
         for token in rawTokens:
-            if ((token.str[0] != '"') and (token.str[0] != '\'')):
+            if (token.str[0] != '"') and (token.str[0] != '\''):
                 continue
-            pos = 1
-            while pos < len(token.str) - 2:
-                pos1 = pos
-                pos = pos + 1
-                if token.str[pos1] != '\\':
-                    continue
-                if token.str[pos1 + 1] == '\\':
-                    pos = pos1 + 2
-                    continue
-                if token.str[pos1 + 1] == 'x':
-                    if not isHexDigit(token.str[pos1 + 2]):
-                        self.reportError(token, 4, 1)
-                        continue
-                    if not isHexDigit(token.str[pos1 + 3]):
-                        self.reportError(token, 4, 1)
-                        continue
-                elif isOctalDigit(token.str[pos1 + 1]):
-                    if not isOctalDigit(token.str[pos1 + 2]):
-                        self.reportError(token, 4, 1)
+            if len(token.str) < 3:
+                continue
+
+            # Examine only assignment operations
```

Comment:
As far as I understand the rule 4.1 it is not only valid for assignment.
The original misra.py also works for this code:
```c
void misra_4_1()
{
    (void)printf("\x41g"); // 4.1
    (void)printf("\x41\x42");
    (void)printf("\x41" "g");
}
```
With this PR the violation is no longer found.
IMHO it would be good to add some non-assignment examples in misra-test.c and make sure they also work.
What do you think?